src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs

```diff
@@ -21,20 +20,17 @@ public CustomTagEntry()
             // used as customer input, constructor without parameters is required.
         }
 
-        public CustomTagEntry(string path, string vR, CustomTagLevel level, CustomTagStatus status)
```

Comment:
Not sure if it's good idea to remove this constructor. My though is that since we already have setter on  all properties, do we still this?

Reply:
There are now three options in C# now for setting data:
- A constructor/ctor
    - Can ensure multiple values are set in the resulting object
    - Can perform complex multi-input validation
    - Can be inherited to enforce/help creation patterns
    - Can enforce readonly semantics
- `public` setters (with optional validation)
    - Useful for "record" types that simply holds data: any combination of the fields/properties can have non-null data
    - Unfortunately the object becomes mutable as the setters must be `public`
- New "`init`-only" setters in C# 9
    - Very much the same as setters, but now the object can be immutable!

In this scenario, if we want to ensure all of the fields (or some subset) must be specified, then we should use a ctor

Reply:
CustomTagEntry is used for API input/output.
* As input, no property is mandatory, and we do validation in https://github.com/microsoft/dicom-server/blob/feature/customtags/src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntryValidator.cs later.
* As output, it's simply mapping to database record, no validation is supposed to be there.
Based on these facts, I guess we can remove the constructor, am I right?